### PR TITLE
Some en-passant proof reading

### PR DESCRIPTION
**Pull request by raoulb**
Proposal for some text changes that occurred to me while reading the docs.

### CODE REVIEW

**Comment by Totktonada** [comment]
Nice! I'll take a look.

**Review by Totktonada — DISMISSED** [review]
LGTM.

**Comment by Totktonada** [comment]
I'll find a second reviewer. Thanks for the patience.

**Comment by Totktonada** [comment]
Maria planned to glance on the patch on the next week.

File: doc/under_the_hood.rst [hunk 1]
```diff
@@ -101,13 +100,13 @@ Iterator Types
 Pure functional iterators
 `````````````````````````
 
-Iterators can be either pure functional or have some side effects and returns
-different values for some initial conditions [#pure_function]_. An **iterator is
+Iterators can be either pure functional or have some side effects and return
+different values for the same initial conditions [#pure_function]_. An **iterator is
 pure functional** if it meets the following criteria:
```
Comment (patiencedaur) [inline doc/under_the_hood.rst:106]:
```suggestion
Iterators can be either purely functional or have some side effects and return
different values for the same initial conditions [#pure_function]_. An **iterator is
purely functional** if it meets the following criteria:

```

"Pure" is an adjective, "purely" is an adverb

File: doc/under_the_hood.rst [hunk 2]
```diff
@@ -136,11 +135,11 @@ machine code and executing them.
 First profiling information for loops is collected. After a hot loop has been
 identified, a special tracing mode is entered which records all executed
 operations of that loop. This sequence of operations is called a **trace**.
-The trace is then optimized and compiled to machine code (trace). When this
+The trace is then optimized and compiled to machine code. When this
 loop is executed again the compiled trace is called instead of the program
 counterpart [#tracing_jit]_.
 
-Why the tracing JIT is important for us? The LuaJIT tracing compiler can detect
+Why is the tracing JIT important for us? The LuaJIT tracing compiler can detect
```
Comment (patiencedaur) [inline doc/under_the_hood.rst:142]:
```suggestion
Why is tracing JIT important for us? The LuaJIT tracing compiler can detect
```

File: doc/under_the_hood.rst [hunk 3]
```diff
@@ -25,23 +25,23 @@ implementation:
...
+   a specific instance of the generating function. For example, the table itself
+   in the ``ipairs`` case.
 
 ``state`` -- third value
    A some transient state of an iterator that is changed after each iteration.
-   For example, an array index for ``ipairs`` case.
+   For example, the array index in the ``ipairs`` case.
 
-Try to call ``gen`` function manually:
+Try to call the ``gen`` function manually:
```
[the remark this answers is not in the document]
Comment (raoulb) [reply doc/under_the_hood.rst:44]: Thanks. I just tried to correct what seemed off to me.

File: doc/under_the_hood.rst [hunk 4]
```diff
@@ -51,11 +51,11 @@ Try to call ``gen`` function manually:
 
 The ``gen`` function returned a new state ``1`` and the next iteration
 value ``a``. The second call to ``gen`` with the new state will return the next
-state  and the next iteration value. When the iterator finishes to the end
+state and the next iteration value. When the iterator finishes to the end
```
[the remark this answers is not in the document]
Comment (raoulb) [reply doc/under_the_hood.rst]: Maybe even start with "If the iterator ..." to account for infinite iterators?

[the remark this answers is not in the document]
Comment (Totktonada) [reply doc/under_the_hood.rst]: Usually we operate on finite iterators, so I would leave 'when' here. Technically 'if' would be more correct, so I don't mind. Up to you.

**Comment by Totktonada** [comment]
Squashed commits, added `doc:` prefix into the commit message to easier distinguish changes of different kinds.

File: doc/under_the_hood.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (Totktonada) [reply doc/under_the_hood.rst]: I'll resolve the discussion, because the main problem was resolved.

**Review by Totktonada — APPROVED** [review]

**Review by patiencedaur — APPROVED** [review]

**Comment by Totktonada** [comment]
Regenerated the documentation: https://luafun.github.io/under_the_hood.html.

**Pull request merged**